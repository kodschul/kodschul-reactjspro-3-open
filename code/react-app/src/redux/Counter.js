import { useSelector, useDispatch } from "react-redux";
import { increment, decrement } from "./slices/counter";

const Counter = () => {
  const count = useSelector((state) => state.counter.count);
  const dispatch = useDispatch();

  const inc = () => dispatch(increment());
  const dec = () => dispatch(decrement());

  console.log("re-render");

  return (
    <>
      <button onClick={inc}>+</button>
      <p>{count}</p>
      <button onClick={dec}>-</button>
    </>
  );
};

export default Counter;
