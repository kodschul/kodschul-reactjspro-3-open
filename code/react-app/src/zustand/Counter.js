import { create } from "zustand";

const useCounter = create((set) => ({
  count: 20,
  inc: (value = 1) =>
    set((state) => ({ ...state, count: state.count + value })),
  dec: (value = 1) =>
    set((state) => ({ ...state, count: state.count - value })),
}));

const Counter = () => {
  const { count, inc, dec } = useCounter();

  return (
    <>
      <button onClick={() => inc(2)}>+</button>
      <p>{count}</p>
      <button onClick={() => dec(2)}>-</button>
    </>
  );
};

export default Counter;
