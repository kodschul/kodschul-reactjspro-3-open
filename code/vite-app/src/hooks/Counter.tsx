import { useState } from "react";
import "../App.css";
import useCounter from "./useCounter";
// import useReactionGame from "./useReactionGame";

type ReactionGameOutput = {
  start: () => void;
  hit: () => void;
  isWaiting: boolean;
  reactionTime?: number;
  isGameStarted: boolean;
};

function App() {
  const { count, inc, dec, reset } = useCounter(0, 5);

  //   const { start, hit, isGameStarted, reactionTime }: ReactionGameOutput =
  //     useReactionGame();

  return (
    <>
      <h1>Counter</h1>
      <div className="card">
        <button
          onClick={inc}

          // onClick={() => setCount((count) => count + 1)}
        >
          count is {count}
        </button>

        <button onClick={dec}>-</button>
        <button onClick={reset}>reset</button>
        <p>This is an Empty App</p>
      </div>
    </>
  );
}

function App2() {
  const { count, inc, dec, reset } = useCounter(0, 10);

  return (
    <>
      <h1>Counter 2</h1>
      <div className="card">
        <button
          onClick={inc}

          // onClick={() => setCount((count) => count + 1)}
        >
          count is {count}
        </button>

        <button onClick={dec}>-</button>
        <button onClick={reset}>reset</button>
        <p>This is an Empty App</p>
      </div>
    </>
  );
}

const AppContainer = () => {
  return (
    <>
      <App />
      <App2 />
    </>
  );
};

export default AppContainer;
