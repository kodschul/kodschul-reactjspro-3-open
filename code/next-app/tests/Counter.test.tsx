import { fireEvent, render, screen } from "@testing-library/react";
import Counter from "../pages/counter";
import { act } from "react";

describe("Counter", () => {
  it.skip("should render the component", () => {
    const { container } = render(<Counter />);
    expect(container).toMatchSnapshot();
  });

  it("should increment the count by 1", () => {
    const { getByTestId } = render(<Counter />);
    const button = screen.getByText(/count is 0/i);

    fireEvent.click(button);

    const btn2 = screen.getByText("count is 1");

    expect(screen.getByText("count is 1")).toBeDefined();
  });
});
