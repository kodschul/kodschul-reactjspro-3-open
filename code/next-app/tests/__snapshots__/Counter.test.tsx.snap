// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render the component 1`] = `
<div>
  <h1>
    Counter
  </h1>
  <div
    class="card"
  >
    <button>
      count is 0
    </button>
    <p>
      This is an Empty App
    </p>
  </div>
</div>
`;
